src/test/java/org/jenkinsci/plugins/badge/EmbeddableBadgeConfigTest.java

```diff
@@ -35,4 +35,66 @@ public void testGetColor(String status, String expected) {
         embeddableBadgeConfig.setStatus(status);
         assertThat(embeddableBadgeConfig.getColor(), is(expected));
     }
+
+    @Test
+    public void testSetSubject() {
+        String subject = "Test Subject";
+        EmbeddableBadgeConfig embeddableBadgeConfig = new EmbeddableBadgeConfig("testId-subject");
+        embeddableBadgeConfig.setSubject(subject);
+        assertThat(embeddableBadgeConfig.getSubject(), is(subject));
+    }
+
+    @Test
+    public void testSetValidLink() {
+        String link = "https://jenkins.io";
+        EmbeddableBadgeConfig config = new EmbeddableBadgeConfig("test-valid-link");
+        config.setLink(link);
+        assertThat(config.getLink(), is(link));
+    }
+
+    @Test
+    public void testSetEmptyLink() {
+        String link = "";
+        EmbeddableBadgeConfig config = new EmbeddableBadgeConfig("test-empty-link");
+        config.setLink(link);
+        assertThat(config.getLink(), is(link));
+    }
+
+    @Test
+    public void testSetValidDefaultColor() {
+        String color = "red";
+        EmbeddableBadgeConfig config = new EmbeddableBadgeConfig("test-valid-default-color");
+        config.setColor(color);
+        assertThat(config.getColor(), is(color));
+    }
+
+    @Test
+    public void testSetValidCustomColor() {
+        String color = "#ff00ff"; // Magenta
+        EmbeddableBadgeConfig config = new EmbeddableBadgeConfig("test-valid-custom-color");
+        config.setColor(color);
+        assertThat(config.getColor(), is(color));
+    }
+
+    @Test
+    public void testSetEmptyColor() {
+        String color = "";
+        EmbeddableBadgeConfig config = new EmbeddableBadgeConfig("test-empty-color");
+        config.setColor(color);
+        assertThat(config.getColor(), is(nullValue()));
```

Comment:
The assertion is incorrect.  Since `color` is the empty string and is not null, the empty string is returned by `getColor`.
```suggestion
        assertThat(config.getColor(), is(""));
```